release/kdm/rke2.go

```diff
@@ -0,0 +1,467 @@
+	if err != nil {
+		return 0, Release{}, err
+	}
+
+	prevReleasePos, err := u.getPreviousReleasePos(prevVersion)
+	if err != nil {
+		return 0, Release{}, err
+	}
+	release := Release{}
+	node := u.releasesSeqNode.Content[prevReleasePos]
+
+	if node.Kind != yaml.MappingNode {
+		return 0, Release{}, fmt.Errorf("not a mapping node in '%s' release", prevVersion)
+	}
+	for i := 0; i < len(node.Content); i += 2 {
+		keyNode := node.Content[i]
+		valueNode := node.Content[i+1]
+		if keyNode.Kind == yaml.ScalarNode {
+			switch keyNode.Value {
+			case "version":
+				release.Version = valueNode.Value
+			case "minChannelServerVersion":
+				release.MinChannelServerVersion = valueNode.Value
+			case "maxChannelServerVersion":
+				release.MaxChannelServerVersion = valueNode.Value
+			case "agentArgs":
+				if valueNode.Kind == yaml.MappingNode && valueNode.Anchor != "" {
+					release.agentArgsAnchor = valueNode.Anchor // This anchor name is from the file, assume it's valid
+					continue
+				}
+				if valueNode.Kind == yaml.AliasNode {
+					release.agentArgsAnchor = valueNode.Value
+					continue
+				}
+			case "serverArgs":
+				if valueNode.Kind == yaml.MappingNode && valueNode.Anchor != "" {
+					release.serverArgsAnchor = valueNode.Anchor // This anchor name is from the file, assume it's valid
+					continue
+				}
+				if valueNode.Kind == yaml.AliasNode {
+					release.serverArgsAnchor = valueNode.Value
+					continue
+				}
+
+			case "featureVersions":
+				if valueNode.Kind == yaml.MappingNode && valueNode.Anchor != "" {
+					release.featureVersionsAnchor = valueNode.Anchor // This anchor name is from the file, assume it's valid
+					continue
+				}
+				if valueNode.Kind == yaml.AliasNode {
+					release.featureVersionsAnchor = valueNode.Value
+					continue
+				}
+			case "charts":
+				if valueNode.Kind == yaml.MappingNode && valueNode.Anchor != "" {
+					release.chartsAnchor = valueNode.Anchor // This anchor name is from the file, assume it's valid
+					continue
+				}
+			}
+		}
+	}
+	return prevReleasePos, release, nil
+}
+
+func (u *RKE2ChannelsUpdater) getPreviousReleasePos(version string) (int, error) {
+	for i := 0; i < len(u.releasesSeqNode.Content); i++ {
+		node := u.releasesSeqNode.Content[i]
+		if node.Kind == yaml.MappingNode {
+			for j := 0; j < len(node.Content); j += 2 {
+				keyNode := node.Content[j]
+				valueNode := node.Content[j+1]
+				if keyNode.Kind == yaml.ScalarNode {
+					switch keyNode.Value {
+					case "version":
+						if valueNode.Value == version {
+							return i, nil
+						}
+					}
+				}
+			}
+		}
+	}
+	return -1, fmt.Errorf("unable to find release '%s'", version)
+}
+
+func (u *RKE2ChannelsUpdater) Bytes() ([]byte, error) {
+	var buf bytes.Buffer
+
+	encoder := yaml.NewEncoder(&buf)
+
+	encoder.SetIndent(2)
+
+	if err := encoder.Encode(&u.rootNode); err != nil {
+		return nil, err
+	}
+
+	if err := encoder.Close(); err != nil {
+		return nil, err
+	}
+
+	outputBytes := buf.Bytes()
```

Comment:
We know the type in the declaration so we can call this variable `output`.

Reply:
done